src/Formatter/RemoveSensitiveQueryStringFormatter.php

```diff
@@ -0,0 +1,51 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Phpro\HttpTools\Formatter;
+
+use Http\Message\Formatter as HttpFormatter;
+use function preg_quote;
+use Psr\Http\Message\RequestInterface;
+use Psr\Http\Message\ResponseInterface;
+use function Safe\preg_replace;
+
+final class RemoveSensitiveDataFormatter implements HttpFormatter
+{
+    /**
+     * @param non-empty-list<string> $sensitiveKeys
+     */
+    public function __construct(
+        private HttpFormatter $formatter,
+        private array $sensitiveKeys
+    ) {
+    }
+
+    public function formatRequest(RequestInterface $request): string
+    {
+        return $this->removeQueryString(
+            $this->formatter->formatRequest($request)
+        );
+    }
+
+    public function formatResponse(ResponseInterface $response): string
+    {
+        return $this->removeQueryString(
+            $this->formatter->formatResponse($response)
+        );
+    }
+
+    private function removeQueryString(string $info): string
+    {
+        foreach ($this->sensitiveKeys as $key) {
+            /** @var string $info */
+            $info = preg_replace(
```

Comment:
You are now trying to replace params in the complete payload.
Meaning that if you send a POST with content-type `application/x-www-form-urlencoded`, it will also replace the parameters in there.

Since this only needs to happen on Requests and requests are immutable, you might be able to :

```php
$uri = $request->getUri();
$this->formatter->format(
    $request->withUri(
        $uri->withQuery(replacemagic($uri->getQuery()));
    )
);
```

Reply:
Thanks, wasn't aware of this behavior!